doc/user/content/transform-data/patterns/percentiles.md

```diff
@@ -10,134 +10,306 @@ menu:
+  precise percentiles. However, calculating distributions using histograms
+  involves a self-join, which can cause memory usage to grow quadratically.  As
+  such, percentile calculations using histograms are better suited for domains
+  with _low_ cardinality.
+
+- [HDR histograms](#using-hdr-histograms-to-compute-approximate-percentiles)
+  further reduce the memory footprint but computes approximate percentiles. HDR
+  histograms are recommended if there is a long tail of large values that you
+  want to track, which is often the case for latency measurements.
 
 ## Using histograms to compute percentiles
 
-Histograms reduce the memory footprint by tracking a count for each unique value, in a `bucket`, instead of tracking all values. Given an `input`, define the histogram for `values` as follows:
+Histograms track the count/frequency for each unique value/bucket instead of
+tracking all values, and can be used to compute exact percentiles. However, for
+domains with high cardinality, because of the cross join to calcuate
+distributions, the memory usage can become significant. As such, using
+histograms to compute percentiles are better suited for domains with _low_
+cardinality.
+
+To use histograms to compute percentiles:
+
+- First, create a histogram view that tracks the unique values and their
+  associated count/frequency.
+
+- Then, using a cross join on the histogram view, create a distribution view
+  that calculates the distribution of the values. The distribution is calculated
+  by dividing the cumulative frequency (the count of all values up to and
+  including that bucket) by the total counts for all bucket values.
+
+  {{< note >}}
+
+  Because of the cross join (which grows quadratic in the number of unique
+  values), the required memory may grow large for domain with high cardinality.
+
+  {{</ note >}}
+
+### Example
+
+1. Create a table `input`:
+
+   ```mzsql
+   CREATE TABLE input (value BIGINT);
+   ```
+
+2. Insert into the `input` table values `1` to `10`.
+
+   ```mzsql
+   INSERT INTO input
+   SELECT n FROM generate_series(1,10) AS n;
+   ```
+
+1. Create a `histogram` view to track unique values from the
+   `input` table and their frequency/counts:
+
+   ```mzsql
+   CREATE VIEW histogram AS
+   SELECT
+     value AS bucket,
+     count(*) AS frequency
+   FROM input
+   GROUP BY value;
+   ```
+
+1. Create a view `distribution` to calculate the cumulative frequency/count and
+   the cumulative distribution. The view definition uses a cross join on the
+   `histogram` view. The distribution is calculated by dividing the cumulative
+   frequency (the count of all values up to and including that bucket) by the
+   total counts for all bucket values.
+
+   ```mzsql
+   CREATE VIEW distribution AS
+   SELECT
+     h.bucket,
+     h.frequency,
+     sum(g.frequency) AS cumulative_frequency,
+     sum(g.frequency) / (SELECT sum(frequency) FROM histogram) AS  cumulative_distribution
```

Comment:
I think perhaps this should be `cumulative_density` rather than `cumulative_distribution`?